chain/jsonrpc-primitives/src/errors.rs

```diff
@@ -34,7 +55,7 @@ impl RpcError {
     ///
     /// Mostly for completeness, doesn't do anything but filling in the corresponding fields.
     pub fn new(code: i64, message: String, data: Option<Value>) -> Self {
-        RpcError { code, message, data }
+        RpcError { code, message, data, error_struct: None }
     }
     /// Create an Invalid Param error.
     pub fn invalid_params(data: impl Serialize) -> Self {
```

Comment:
Let's update it to return `ValidationError`.

Also, update `server_error` to return `InternalError`